sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileSasClientTests.groovy

```diff
@@ -116,8 +116,12 @@ class FileSasClientTests extends APISpec {
             .setId("0000")
             .setAccessPolicy(new ShareAccessPolicy().setPermissions("rcwdl")
                 .setExpiresOn(getUTCNow().plusDays(1)))
+
         primaryShareClient.setAccessPolicy(Arrays.asList(identifier))
 
+        // Sleep 30 seconds if running against the live service as it may take ACLs that long to take effect.
+        sleepIfLive(30000)
```

Comment:
Can we use polling with shorter sleeps instead of sleeping for a fixed 30 seconds?  Large, fixed sleeps like this are bad because they make tests slower than necessary and still aren't guaranteed to be reliable.

Reply:
This should be do-able with shorter sleeps then poll, mind writing an issue for it as this is a commonly used pattern across the other modules.

Reply:
Tracking Issue: https://github.com/Azure/azure-sdk-for-java/issues/7353

I am OK with the sleep in this PR since we are already doing it in other tests.  However, I think we should use the same name and code as the existing helper method `sleepIfRecord()`, instead of adding a new helper method with a different name and implementation.